src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -70,7 +74,9 @@ internal sealed class ActiveSolutionBoundTracker : IActiveSolutionBoundTracker,
             this.gitEventsMonitor = gitEventsMonitor;
             this.logger = logger;
 
-            vsMonitorSelection = host.GetService<SVsShellMonitorSelection, IVsMonitorSelection>();
+            // TODO - MEF-ctor should be free-threaded -> we might be on a background thread ->
+            // calling serviceProvider.GetService is dangerous.
+            vsMonitorSelection = serviceProvider.GetService<SVsShellMonitorSelection, IVsMonitorSelection>();
```

Comment:
I've manually debugged to check this class and the other modified ones are created correctly at runtime.